### PR TITLE
*: add web UI

### PR DESCRIPTION
**Pull request by nolouch**
Signed-off-by: nolouch <nolouch@gmail.com>

### What problem does this PR solve? 
support build PD binary with UI.
```
PD_WEB=1 make
```
browse [localhotst:2379/web](http://localhotst:2379/web) to see the ui.
### What is changed and how it works?
add scripts.

### Check List 

Tests 
 - Manual test (add detailed scripts or steps below)

### CODE REVIEW

File: tools.json [hunk 1]
```diff
@@ -19,7 +19,15 @@
...
+      "Commit": "f6a432f39fbebd7aaaf4f29ca1aac539edbf6834"
+    },
+    {
+      "Repository": "gopkg.in/alecthomas/gometalinter.v2",
+      "Commit": "46cc1ea3778b247666c2949669a3333c532fa9c6"
     }
   ],
   "RetoolVersion": "1.3.7"
-}
+}
```
Comment (lhy1024) [inline tools.json]: no newline at end of file

File: server/server.go [hunk 2]
```diff
@@ -62,6 +63,7 @@ const (
 	// pdRootPath for all pd servers.
 	pdRootPath      = "/pd"
 	pdAPIPrefix     = "/pd/"
+	uiPath          = "/ui/"
```
Comment (lhy1024) [inline server/server.go]: how about root path?

Comment (nolouch) [reply server/server.go]: root path already registered :(

Comment (disksing) [reply server/server.go]: How about `/web`?

File: scripts/build-ui.sh [hunk 3]
```diff
@@ -0,0 +1,15 @@
+#!/usr/bin/env bash
+git submodule update --remote
+BASEDIR=$(git rev-parse --show-toplevel)
+UI_DIR="$BASEDIR/pkg/ui/pd-web"
+TARGET_DIST="$BASEDIR/pkg/ui/dist"
+SCRIPTS_DIR="$BASEDIR/scripts"
+echo "##### Build web UI"
+cd $UI_DIR
+yarn
+PUBLIC_URL=/ui/ yarn build
```
Comment (HunDunDM) [inline scripts/build-ui.sh]:
```suggestion
PUBLIC_URL=/ui yarn build
```

Not recommended to end with `/`

File: server/server.go (hunk 2 above)
Comment (HunDunDM) [reply server/server.go]: How about `/gui`?

Comment (lhy1024) [reply server/server.go]: I think `/web` is better

Comment (nolouch) [reply server/server.go]: I use `/web` finally.

File: pkg/ui/ui.go [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+
+// Handler returns an http.Handler that serves the UI.
+func Handler() http.Handler {
+	fileServer := http.FileServer(&assetfs.AssetFS{
+		Asset:     Asset,
+		AssetDir:  AssetDir,
+		AssetInfo: AssetInfo,
+	})
+
+	haveWebUI := Asset != nil && AssetDir != nil && AssetInfo != nil
```
Comment (lhy1024) [inline pkg/ui/ui.go]: how about `enableWebUI`?

**Pull request merged**